packages/vite/types/importMeta.d.ts

```diff
@@ -59,8 +59,14 @@ interface ImportMeta {
   >
 }
 
-interface ImportMetaEnv {
+// The index signature is for user-defined variables, which are always strings. Some
+// predefined properties are boolean, and TypeScript doesn't allow them to have types
+// that don't match the index signature. However, in this case properties are always
+// accessed statically (.PROP, not [PROP]), so the limitation doesn't matter.
+// The intersection type is a workaround.
+type ImportMetaEnv = {
   [key: string]: string | undefined
+} & {
```

Comment:
Did you try `interface ImportMetaEnv extends Readonly<Record<string, string>> { ...`?

Reply:
Just tried, produces the same error as the plain interface with an index signature.

Reply:
Maybe do `Record<string, string | undefined> & {` instead.

And it doesn't need the comment, IMO. If someone tries to change it to an interface, they'll get the TypeScript error.

Reply:
Done.